packages/builder/src/components/automation/SetupPanel/RowSelector.svelte

```diff
@@ -1,61 +1,163 @@
 <script>
   import { tables } from "stores/builder"
-  import { Select, Checkbox, Label } from "@budibase/bbui"
+  import {
+    ActionButton,
+    Popover,
+    Icon,
+    TooltipPosition,
+    TooltipType,
+  } from "@budibase/bbui"
   import { createEventDispatcher } from "svelte"
   import { FieldType } from "@budibase/types"
 
   import RowSelectorTypes from "./RowSelectorTypes.svelte"
   import DrawerBindableSlot from "../../common/bindings/DrawerBindableSlot.svelte"
   import AutomationBindingPanel from "../../common/bindings/ServerBindingPanel.svelte"
-  import { TableNames } from "constants"
+  import { FIELDS } from "constants/backend"
+  import { capitalise } from "helpers"
+  import { memo } from "@budibase/frontend-core"
+  import PropField from "./PropField.svelte"
+  import { cloneDeep, isPlainObject, mergeWith } from "lodash"
 
   const dispatch = createEventDispatcher()
-  export let value
+
+  export let row
   export let meta
   export let bindings
   export let isTestModal
-  export let isUpdateRow
 
-  $: parsedBindings = bindings.map(binding => {
-    let clone = Object.assign({}, binding)
-    clone.icon = "ShareAndroid"
-    return clone
+  const memoStore = memo({
+    row,
+    meta,
   })
 
   let table
+  // Row Schema Field
   let schemaFields
   let attachmentTypes = [
     FieldType.ATTACHMENTS,
     FieldType.ATTACHMENT_SINGLE,
     FieldType.SIGNATURE_SINGLE,
   ]
 
-  $: {
-    table = $tables.list.find(table => table._id === value?.tableId)
+  let customPopover
+  let popoverAnchor
+  let editableRow = {}
+  let editableFields = {}
 
-    // Just sorting attachment types to the bottom here for a cleaner UX
-    schemaFields = Object.entries(table?.schema ?? {}).sort(
-      ([, schemaA], [, schemaB]) =>
-        (schemaA.type === "attachment") - (schemaB.type === "attachment")
-    )
+  // Avoid unnecessary updates
+  $: memoStore.set({
+    row,
+    meta,
+  })
 
-    schemaFields.forEach(([, schema]) => {
-      if (!schema.autocolumn && !value[schema.name]) {
-        value[schema.name] = ""
-      }
-    })
+  $: if ($memoStore?.meta?.fields) {
+    editableFields = cloneDeep($memoStore?.meta?.fields)
   }
-  const onChangeTable = e => {
-    value["tableId"] = e.detail
-    dispatch("change", value)
+
+  $: parsedBindings = bindings.map(binding => {
+    let clone = Object.assign({}, binding)
+    clone.icon = "ShareAndroid"
+    return clone
+  })
+
+  $: tableId = $memoStore?.row?.tableId
+  $: if (tableId) {
```

Comment:
This is a pretty huge reactive statment! Looks like it will react to changes in `tableId`, `memoStore`, `editableRow` and `editableFields` because they are used as values too.

I would definitely refactor this to be a function, with obvious props passed in that we want to react to. It would be really hard in future to work with this otherwise, and there are so many potential triggers. This also has the side effect of mutating `editableFields` many times, whereas we could do it in a single update if we used a function.